File: NavApp/app/src/main/java/com/example/nav/Entities/Goods.kt
package com.example.nav.Entities

import com.google.gson.annotations.SerializedName

class   Goods {
    @SerializedName("id")
    var id: Int = 0

    @SerializedName("title")
    var title: String? = ""

    @SerializedName("description")
    var description: String? = ""

    @SerializedName("phone")
    var phone: String? = ""

    @SerializedName("address")
    var address: String? = ""

    @SerializedName("type")
    var type: ArrayList<Types_Goods> = ArrayList()

    @SerializedName("seller")
    var seller: String? = ""

    @SerializedName("userId")
    var userId: Int? = 0

    @SerializedName("images")
    var images: ArrayList<Image> = ArrayList()
}

File: shopApi/goodsShop/Models/Goods.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace goodsShop.Models
{
    public class Goods
    {
        public long id { get; set; }
        public string title { get; set; }
        public string description { get; set; }
        public string phone { get; set; }
        public string address { get; set; }
        public List<Goods_Type> type { get; set; }
        public string seller { get; set; }
        public long userId { get; set; }
        public List<Image> images { get; set; } //list of byte arrays, one image - one byte array
    }
}
